### PR TITLE
entity:delete -> execute deletion of entities in small steps to avoid loading all at once

### PR DESCRIPTION
**Pull request by cgoffin**
When deleting lots of entities at once (+1000). It seems that the command hangs. I changed the command to process the deletions in small steps and added a progress bar to the output to inform the user.

### CODE REVIEW

File: src/Drupal/Commands/core/EntityCommands.php [hunk 1]
```diff
@@ -40,18 +44,22 @@ public function __construct(EntityTypeManagerInterface $entityTypeManager)
...
      *   Delete all users except uid=1.
+     * @usage drush entity:delete node --chunks=5
+     *   Delete all node entities in steps of 5.
      *
      * @command entity:delete
      * @aliases edel,entity-delete
      * @throws \Exception
      */
-    public function delete($entity_type, $ids = null, $options = ['bundle' => self::REQ, 'exclude' => self::REQ])
+    public function delete(OutputInterface $output, $entity_type, $ids = null, $options = ['bundle' => self::REQ, 'exclude' => self::REQ, 'chunks' => self::REQ])
```
Comment (weitzman) [inline src/Drupal/Commands/core/EntityCommands.php:54]: I didn't know that OutputInterface would work here. Look kosher to you, @greg-1-anderson? 


Also, lets just default chunks to 50 or so. We get advantages to bulk deleting so I think 10 is too few.

**Review by weitzman — COMMENTED** [review]
Great work.

File: src/Drupal/Commands/core/EntityCommands.php (hunk 1 above)
Comment (cgoffin) [reply src/Drupal/Commands/core/EntityCommands.php:54]: I altered the default to 50 and fixed some coding standard issues. Why do you think the OutputInterface wouldn't work? I tested on my Drupal installation and it worked fine.

**Comment by weitzman** [comment]
Thanks. This is our first progress bar in core.

I think its a. bit cleaner if we use $this->io()->createProgressBar() and $this->io()->progressAdvance(). Could you try that?

**Comment by cgoffin** [comment]
I adjusted the code and on my local Drupal environment it worked.

File: src/Drupal/Commands/core/EntityCommands.php [hunk 2]
```diff
@@ -60,23 +64,42 @@ public function delete($entity_type, $ids = null, $options = ['bundle' => self::
...
+            unset($result[0], $result[1]);
         }
 
-        if (empty($entities)) {
+        if (empty($result)) {
             $this->logger()->success(dt('No matching entities found.'));
         } else {
-            $storage->delete($entities);
-            $this->logger()->success(dt('Deleted !type entity Ids: !ids', ['!type' => $entity_type, '!ids' => implode(', ', array_keys($entities))]));
+            $chunks = $options['chunks'] ?? 50;
```
Comment (weitzman) [inline src/Drupal/Commands/core/EntityCommands.php:78]: Lets remove this line, and just default to 50 in the method params. That way the default is documented in help.

**Review by weitzman — CHANGES_REQUESTED** [review]
Looking good. One minor comment below.

File: src/Drupal/Commands/core/EntityCommands.php (hunk 2 above)
Comment (cgoffin) [reply src/Drupal/Commands/core/EntityCommands.php]: Did know that, nice. I did the adjustment.

**Comment by weitzman** [comment]
Thanks @cgoffin.

**Pull request merged**